torch/distributed/distributed_c10d.py

```diff
@@ -1916,7 +1916,7 @@ def destroy_process_group(group: Optional[ProcessGroup] = None):
     # alive until all works and hooks are done. The current implementation does the
     # latter. Therefore, we explicitly call _wait_for_pending_works() here to wait
     # for the pending hooks to finish.
-    if pg.name().lower() == "nccl" and pg._has_hooks():
+    if pg._has_hooks():
```

Comment:
Is there some way to ensure the backend supports batch and use that property of the backend?

Does your backend implement ProcessGroupNCCL or does it provide a new c10d backend?

Reply:
> Is there some way to ensure the backend supports batch and use that property of the backend?
> 
> Does your backend implement ProcessGroupNCCL or does it provide a new c10d backend?

1. If pg._has_hooks() return True,It means that pg._register_on_completion_hook() has already been called, and pg. _wait_for_pending_works()  also needs to be called.
2. My backend provide a new c10d backend like ProcessGroupNCCL.